### DISCUSSION TITLE
healthcheck problem

### DISCUSSION THREAD

**Discussion by Caibreeze0908** [Kong/kong#10249] [Help] [answered]
log:
2020/12/29 04:21:59 [error] 94#0: *4972303083 [lua] events.lua:364: poll(): worker-events: dropping event; waiting for event data timed out, id: 53826839, context: ngx.timer
2020/12/29 04:22:01 [error] 116#0: *4972304544 [lua] events.lua:364: poll(): worker-events: dropping event; waiting for event data timed out, id: 53826839, context: ngx.timer
2020/12/29 04:22:01 [error] 106#0: *4972303195 [lua] events.lua:364: poll(): worker-events: dropping event; waiting for event data timed out, id: 53826839, context: ngx.timer
2020/12/29 04:22:02 [error] 120#0: *4972290367 [lua] events.lua:364: poll(): worker-events: dropping event; waiting for event data timed out, id: 53826839, context: ngx.timer
2020/12/29 04:22:02 [error] 97#0: *4972306678 [lua] events.lua:364: poll(): worker-events: dropping event; waiting for event data timed out, id: 53826839, context: ngx.timer
2020/12/29 04:22:02 [error] 118#0: *4972303883 [lua] events.lua:364: poll(): worker-events: dropping event; waiting for event data timed out, id: 53826839, context: ngx.timer
2020/12/29 04:22:02 [error] 89#0: *4972304061 [lua] events.lua:364: poll(): worker-events: dropping event; waiting for event data timed out, id: 53826839, context: ngx.timer
2020/12/29 04:22:03 [error] 95#0: *4972289737 [lua] events.lua:364: poll(): worker-events: dropping event; waiting for event data timed out, id: 53826839, context: ngx.timer
2020/12/29 04:22:04 [error] 117#0: *4972289032 [lua] events.lua:364: poll(): worker-events: dropping event; waiting for event data timed out, id: 53826839, context: ngx.timer
2020/12/29 04:22:04 [error] 99#0: *4972289027 [lua] events.lua:364: poll(): worker-events: dropping event; waiting for event data timed out, id: 53826839, context: ngx.timer
2020/12/29 04:22:05 [error] 107#0: *4972289088 [lua] events.lua:364: poll(): worker-events: dropping event; waiting for event data timed out, id: 53826839, context: ngx.timer
2020/12/29 04:22:07 [error] 110#0: *4972302335 [lua] events.lua:364: poll(): worker-events: dropping event; waiting for event data timed out, id: 53826839, context: ngx.timer
2020/12/29 04:22:07 [error] 98#0: *4972289070 [lua] events.lua:364: poll(): worker-events: dropping event; waiting for event data timed out, id: 53826839, context: ngx.timer
2020/12/29 04:22:08 [error] 115#0: *4972288071 [lua] events.lua:364: poll(): worker-events: dropping event; waiting for event data timed out, id: 53826839, context: ngx.timer
2020/12/29 04:22:21 [error] 92#0: *4972291478 [lua] events.lua:364: poll(): worker-events: dropping event; waiting for event data timed out, id: 53826839, context: ngx.timer
2020/12/29 08:03:03 [error] 108#0: *5018542801 [lua] healthcheck.lua:1104: log(): [healthcheck] (1bdcdd5c-ecbc-4cb3-b271-c5c4a3e03f56:ae-app-56.www.hba.main) failed to release lock 'lua-resty-healthcheck:1bdcdd5c-ecbc-4cb3-b271-c5c4a3e03f56:ae-app-56.www.hba.main:target_list_lock': unlocked, context: ngx.timer
2020/12/29 08:45:09 [error] 99#0: *5027657771 [lua] healthcheck.lua:1104: log(): [healthcheck] (1bdcdd5c-ecbc-4cb3-b271-c5c4a3e03f56:ae-app-56.www.hba.main) failed to release lock 'lua-resty-healthcheck:1bdcdd5c-ecbc-4cb3-b271-c5c4a3e03f56:ae-app-56.www.hba.main:target_list_lock': unlocked, context: ngx.timer
2020/12/29 09:06:45 [error] 114#0: *5032074615 [lua] healthcheck.lua:1104: log(): [healthcheck] (1bdcdd5c-ecbc-4cb3-b271-c5c4a3e03f56:ae-app-56.www.hba.main) failed to release lock 'lua-resty-healthcheck:1bdcdd5c-ecbc-4cb3-b271-c5c4a3e03f56:ae-app-56.www.hba.main:target_list_lock': unlocked, context: ngx.timer
** [lua] events.lua:364: poll(): worker-events: dropping event; waiting for event data timed out, id: 53826839, context: ngx.timer**
** failed to release lock 'lua-resty-healthcheck:1bdcdd5c-ecbc-4cb3-b271-c5c4a3e03f56:ae-app-56.www.hba.main:target_list_lock': unlocked, context: ngx.timer **
how to solve these two problem?

**Comment by locao** [comment, marked as answer]
Hi @Caibreeze0908!

This seems an old Kong version. These issues have been fixed in (not so) recent releases. Can you upgrade?

**Reply by Caibreeze0908** [reply]
not update.Can you send a log of the fix or what version it is?

**Reply by locao** [reply]
- [Use the new timer library to improve performance.](https://github.com/Kong/kong/blob/3.0.0/CHANGELOG.md#performance)
- [Introduced unix domain socket based `lua-resty-events` to replace shared memory based `lua-resty-worker-events`.](https://github.com/Kong/kong/blob/3.0.0/CHANGELOG.md#core-1)

**Reply by Caibreeze0908** [reply]
Thanks

**Comment by chronolaw** [comment]
This [blog](https://konghq.com/blog/nginx-openresty-event-handling-strategy-for-cpu-efficiency) describes the new event propagation mechanism.
